### ISSUE TITLE
[Bug] radeon 5300m is not showing up

### ISSUE THREAD

**Issue by Icarus-Xu** [exelban/stats#173]
**Describe the bug**
cannot monitor amd graphic card, the app only shows intel uhd 630

<img width="1721" alt="Screen Shot 2020-11-19 at 12 31 27 PM" src="https://user-images.githubusercontent.com/11155311/99621630-5c15e600-2a63-11eb-8235-bb90176d8389.png">

**Details:**
 - Device: [Macbook Pro 16inch late 2019]
 - MacOS [11.1 Beta 20C5048k]
 - Application version [2.4.3]

**Comment by Icarus-Xu** [comment]
<img width="1722" alt="Screen Shot 2020-11-19 at 12 34 08 PM" src="https://user-images.githubusercontent.com/11155311/99621709-8d8eb180-2a63-11eb-9033-3ad7328dad8f.png">


as you can see in the "gpu to show" list, there is only intel 630

**Comment by Military-Doctor** [comment]
I am the same model as you and have the same problem.

I read the code and found that the program can get two graphics cards properly. 

I am still investigating the specific reason about why the AMD graphics card cannot be displayed.

**Comment by Military-Doctor** [comment]
Maybe you can disable the `Automatic graphics switching` to let your AMD 5300M enabled.

After the graphics card working mode is set to `Discrete Only`, the AMD card will show up.

<img width="326" alt="image" src="https://user-images.githubusercontent.com/14367694/99623157-71404400-2a66-11eb-991c-7da29a3ed32d.png">

<img width="757" alt="image" src="https://user-images.githubusercontent.com/14367694/99623209-8f0da900-2a66-11eb-8518-1505ad362fbb.png">

**Comment by Icarus-Xu** [comment]
thanks for replying, but after i turned off auto graphic switching, this "gpu to show" list is simply empty :( I'm not sure what's wrong.

**Comment by Icarus-Xu** [comment]
@Military-Doctor

**Comment by Icarus-Xu** [comment]
ah ok, I got it right. but still the temperature is not correct on radeon, not sure if it has a sensor and can read its value

**Comment by exelban** [comment]
Hi. Thanks for reporting.
I suppose the temperature is not available, because of the driver not providing it.
Could you please post `ioreg -r -d 1 -w 0 -c "IOAccelerator" > output.txt`?

**Comment by Icarus-Xu** [comment]
```
+-o AMDRadeonX6000_AMDNavi14GraphicsAccelerator  <class AMDRadeonX6000_AMDNavi14GraphicsAccelerator, id 0x100000727, registered, matched, active, busy 0 (209 ms), retain 113>
    {
      "AccelCaps" = 15
      "IODVDBundleName" = "AMDRadeonVADriver2"
      "IOGVAHEVCEncode" = "1"
      "ATY,intrev" = 21
      "AccelNativeDMARowByteAlignment" = 256
      "InternalStatisticsAccm" = {}
      "MetalStatisticsName" = "GFX10Statistics"
      "GPURawCounterBundleName" = "AMDRawCounterPlugin"
      "GPUDCCDisplayable" = Yes
      "IOGeneralInterest" = "IOCommand is not serializable"
      "InternalStatistics" = {}
      "MetalPluginClassName" = "AMDMTLGFX10Device"
      "GPURawCounterPluginClassName" = "AMDRawCounterSourceGroup"
      "GpuDebugPolicy" = 0
      "AAPL,OpenCLdisabled" = Yes
      "AGCInfo" = {"fBusyCount"=0,"poweredOffByAGC"=Yes,"fSubmissionsSinceLastCheck"=0,"fLastSubmissionPID"=1482}
      "dpm" = 1
      "MetalPluginName" = "AMDRadeonX6000MTLDriver"
      "PerformanceStatisticsAccum" = {"orphanedReusableVidMemoryHitRate"=87,"texturePageOutBytes"=0,"swapCompleteVideoWaitTime"=0,"orphanedReusableSysMemoryHitRate"=0,"dataBufferCount"=0,"oolTexturePageInBytes"=0,"textureCount"=1563,"iosurfaceTextureCreationCount"=0,"surfaceReadLockIdleWaitTime"=0,"freeDataBufferWaitTime"=0,"orphanedReusableSysMemoryBytes"=0,"ioSurfaceReadOutBytes"=0,"agprefTextureCreationCount"=0,"orphanedReusableSysMemoryCount"=0,"finishAll2DWaitTime"=0,"orphanedNonReusableSysMemoryCount"=0,"ioSurfacePageOutBytes"=0,"surfaceBufferReadOutBytes"=0,"surfaceTextureCreationCount"=0,"textureReadOutBytes"=0,"gartMapInBytesPerSample"=0,"swapBytesPerSample"=0,"dataBytesPerSample"=0,"surfaceCount"=117,"clientGLWaitTime"=0,"surfaceCopyInWaitTime"=0,"clientSharedAllocatedBytes"=1224704,"surfaceCopyOutWaitTime"=0,"gartMapOutBytesPerSample"=0,"ioSurfacePageInBytes"=0,"gartCacheBytes"=33554432,"volatileSurfaceCount"=0,"vramEvictionWaitTime"=0,"oolTextureCreationBytes"=0,"freeToAllocGPUAddressWaitTime"=0,"oolTextureCreationCount"=0,"orphanedReusableVidMemoryCount"=0,"orphanedReusableVidMemoryBytes"=0,"context2DCount"=5,"inUseSysMemoryBytes"=3936256,"bufferSwapCount"=0,"orphanedNonReusableVidMemoryBytes"=0,"contextGLCount"=4,"finish2DWaitTime"=0,"gartSizeBytes"=6442450944,"recoveryCount"=0,"stdTexturePageInBytes"=0,"agprefTextureCreationBytes"=0,"gartUsedBytes"=3936256,"surfaceWriteLockIdleWaitTime"=0,"surfaceBufferPageInBytes"=0,"finishGLWaitTime"=0,"iosurfaceTextureCreationBytes"=0,"finishCLWaitTime"=0,"finishVideoWaitTime"=0,"contextCLCount"=0,"freeSurfaceBackingWaitTime"=0,"swapCompleteGLWaitTime"=0,"orphanedNonReusableSysMemoryBytes"=0,"inUseVidMemoryBytes"=0,"agpTextureCreationBytes"=0,"stdTextureCreationBytes"=0,"agpTextureCreationCount"=0,"stdTextureCreationCount"=0,"surfaceSetShapeIdleWaitTime"=0,"gartFreeBytes"=6438514688,"textureVolunteerUnloadBytes"=0,"bufferFlipCount"=0,"hardwareSubmitWaitTime"=0,"swapComplete2DWaitTime"=0,"surfaceBufferPageOutBytes"=0,"surfaceBufferTextureCreationCount"=0,"hardwareWaitTime"=0,"contextVideoCount"=0,"freeSurfaceSwapBufferWaitTime"=0,"orphanedNonReusableVidMemoryCount"=0}
      "PerformanceStatistics" = {"orphanedReusableVidMemoryHitRate"=87,"texturePageOutBytes"=0,"swapCompleteVideoWaitTime"=0,"orphanedReusableSysMemoryHitRate"=0,"dataBufferCount"=0,"oolTexturePageInBytes"=0,"textureCount"=1563,"iosurfaceTextureCreationCount"=0,"surfaceReadLockIdleWaitTime"=0,"freeDataBufferWaitTime"=0,"orphanedReusableSysMemoryBytes"=0,"ioSurfaceReadOutBytes"=0,"agprefTextureCreationCount"=0,"orphanedReusableSysMemoryCount"=0,"finishAll2DWaitTime"=0,"orphanedNonReusableSysMemoryCount"=0,"ioSurfacePageOutBytes"=0,"surfaceBufferReadOutBytes"=0,"surfaceTextureCreationCount"=0,"textureReadOutBytes"=0,"gartMapInBytesPerSample"=0,"swapBytesPerSample"=0,"dataBytesPerSample"=0,"surfaceCount"=117,"clientGLWaitTime"=0,"surfaceCopyInWaitTime"=0,"clientSharedAllocatedBytes"=1224704,"surfaceCopyOutWaitTime"=0,"gartMapOutBytesPerSample"=0,"ioSurfacePageInBytes"=0,"gartCacheBytes"=33554432,"volatileSurfaceCount"=0,"vramEvictionWaitTime"=0,"oolTextureCreationBytes"=0,"freeToAllocGPUAddressWaitTime"=0,"oolTextureCreationCount"=0,"orphanedReusableVidMemoryCount"=0,"orphanedReusableVidMemoryBytes"=0,"context2DCount"=5,"inUseSysMemoryBytes"=3936256,"bufferSwapCount"=0,"orphanedNonReusableVidMemoryBytes"=0,"contextGLCount"=4,"finish2DWaitTime"=0,"gartSizeBytes"=6442450944,"recoveryCount"=0,"stdTexturePageInBytes"=0,"agprefTextureCreationBytes"=0,"gartUsedBytes"=3936256,"surfaceWriteLockIdleWaitTime"=0,"surfaceBufferPageInBytes"=0,"finishGLWaitTime"=0,"iosurfaceTextureCreationBytes"=0,"finishCLWaitTime"=0,"finishVideoWaitTime"=0,"contextCLCount"=0,"freeSurfaceBackingWaitTime"=0,"swapCompleteGLWaitTime"=0,"orphanedNonReusableSysMemoryBytes"=0,"inUseVidMemoryBytes"=0,"agpTextureCreationBytes"=0,"stdTextureCreationBytes"=0,"agpTextureCreationCount"=0,"stdTextureCreationCount"=0,"surfaceSetShapeIdleWaitTime"=0,"gartFreeBytes"=6438514688,"textureVolunteerUnloadBytes"=0,"bufferFlipCount"=0,"hardwareSubmitWaitTime"=0,"swapComplete2DWaitTime"=0,"surfaceBufferPageOutBytes"=0,"surfaceBufferTextureCreationCount"=0,"hardwareWaitTime"=0,"contextVideoCount"=0,"freeSurfaceSwapBufferWaitTime"=0,"orphanedNonReusableVidMemoryCount"=0}
      "IOGLBundleName" = "AMDRadeonX6000GLDriver"
      "IOPropertyMatch" = {"LoadAccelerator"=Yes}
      "IOKitDebug" = 0
      "IOSourceVersion" = "0.0.0.0.0"
      "IOPersonalityPublisher" = "com.apple.kext.AMDRadeonX6000"
      "GPUDebugNullClientMask" = 0
      "IOGVAHEVCDecodeCapabilities" = {"VTPerProfileDetails"={"1"={"VTMaxDecodeLevel"=153},"2"={"VTMaxDecodeLevel"=153},"3"={"VTMaxDecodeLevel"=153}},"VTSupportedProfileArray"=(1,2,3)}
      "IOGVAH264EncodeCapabilities" = {"VTRating"=350,"VTQualityRating"=50}
      "ATY,cbits" = 262144
      "IOOCDBundleName" = "AMDRadeonX6000GLDriver"
      "cail_properties" = {}
      "CFBundleIdentifier" = "com.apple.kext.AMDRadeonX6000"
      "IOAccelRevision" = 2
      "IOVARendererID" = 16908292
      "IOProviderClass" = "IOPCIDevice"
      "IOCFPlugInTypes" = {"ACCF0000-0000-0000-0000-000a2789904e"="IOAccelerator2D.plugin"}
      "IOGVACodec" = "AMDVCN2"
      "SWIP_properties" = {"EnableSwVCNFWLoading"=0}
      "IOGVAHEVCDecode" = "1"
      "IOClass" = "AMDRadeonX6000_AMDNavi14GraphicsAccelerator"
      "CFBundleIdentifierKernel" = "com.apple.kext.AMDRadeonX6000"
      "IOReportLegendPublic" = Yes
      "GPUConfigurationVariable" = {"NumROP"=8,"MemBusWidth"=128,"NumCUPerSH"=10,"NumSH"=2,"SysClkFreq"=125000,"NumSE"=1,"MemClkFreq"=73600}
      "IOPCITunnelCompatible" = Yes
      "IOPCIMatch" = "0x73401002 0x73411002 0x73431002 0x73471002 0x734F1002"
      "IOMatchCategory" = "IOAccelerator"
      "IOReportLegend" = ({"IOReportChannels"=((4000,6442450945,"Overall GPU Busy"),(4001,6442450945,"Overall GPU total time"),(4002,6442450945,"3d Busy"),(4003,6442450945,"3d total time"),(4010,6442450945,"DMA Busy"),(4011,6442450945,"DMA total time"),(4006,6442450945,"Video Busy"),(4007,6442450945,"Video total time"),(4008,6442450945,"Compute Busy"),(4009,6442450945,"Compute total time"),(4705492258585592320,6442450945,"GFX busy"),(4705492258585596928,6442450945,"GFX total time"),(4705492241555999810,6442450945,"Compute low priority busy"),(4705492241555999828,6442450945,"Compute low priority total time"),(4705492241556000066,6442450945,"Compute medium priority busy"),(4705492241556000084,6442450945,"Compute medium priority total time"),(4705492241555998786,6442450945,"Compute high priority busy"),(4705492241555998804,6442450945,"Compute high priority total time"),(4705492245816619074,6442450945,"DMA 0 busy"),(4705492245816619092,6442450945,"DMA 0 total time"),(4705492245816619330,6442450945,"DMA 1 busy"),(4705492245816619348,6442450945,"DMA 1 total time"),(0,6442450945,"Command Buffer Bytes"),(1,6442450945,"GL Command Buffer Bytes"),(2,6442450945,"CL Command Buffer Bytes"),(3,6442450945,"2D Command Buffer Bytes"),(4,6442450945,"Video Command Buffer Bytes"),(5,6442450945,"Out-Of-Line Texture Creation Bytes"),(6,6442450945,"Shared System Memory Texture Creation Bytes"),(7,6442450945,"Private Video Memory Texture Creation Bytes"),(8,6442450945,"Private System Memory Texture Creation Bytes"),(9,6442450945,"IOSurface Backed Texture Creation Bytes"),(10,6442450945,"Out-Of-Line Texture Creation Count"),(11,6442450945,"Private Video Memory Texture Creation Count"),(12,6442450945,"Private System Memory Texture Creation Count"),(13,6442450945,"Shared System Memory Texture Creation Count"),(14,6442450945,"IOAccelSurfaceRef Texture Creation Count"),(15,6442450945,"IOAccelSurfaceBufferRef Texture Creation Count"),(16,6442450945,"IOSurface Backed Texture Creation Count"),(17,6442450945,"Texture Count"),(20,6442450945,"Bytes mapped into the GPU"),(21,6442450945,"Bytes unmapped from the GPU"),(82,6442450945,"IOAccelClientShared allocated bytes"),(22,6442450945,"Private textures bytes paged on"),(23,6442450945,"Out of line texture bytes paged on"),(24,6442450945,"IOAccelSurface bytes paged on"),(25,6442450945,"IOSurface bytes paged on"),(26,6442450945,"IOSurface bytes evicted"),(27,6442450945,"IOAccelSurface bytes evicted"),(28,6442450945,"Texture bytes evicted"),(29,6442450945,"IOSurface bytes paged off"),(30,6442450945,"IOAccelSurface bytes paged off"),(31,6442450945,"Texture bytes paged off"),(32,6442450945,"Texture bytes voluntarily paged off"),(33,6442450945,"IOAccelSurface count"),(34,6442450945,"Volatile IOAccelSurface count"),(35,6442450945,"2D context count"),(36,6442450945,"Video context count"),(37,6442450945,"OpenGL context count"),(38,6442450945,"OpenCL context count"),(39,6442450945,"Buffer swap count"),(40,6442450945,"Buffer flip count"),(41,6442450945,"Databuffer count"),(42,6442450945,"VRAM eviction wait time"),(43,6442450945,"CPU wait on GPU"),(44,6442450945,"CPU wait to submit to GPU"),(45,6442450945,"GL Client wait on GPU"),(46,6442450945,"CL Client wait on GPU"),(47,6442450945,"CPU Wait to evict resource"),(49,6442450945,"CPU wait for free databuffer"),(53,6442450945,"GL Client wait for swap complete"),(55,6442450945,"CPU Wait to copy in surface"),(56,6442450945,"CPU Wait to copy out surface"),(59,6442450945,"CPU Wait for surface idle"),(60,6442450945,"CPU Wait for resource read"),(61,6442450945,"CPU Wait for resource write"),(62,6442450945,"CPU Wait for all 2D finish"),(63,6442450945,"CPU Wait for 2D finish"),(64,6442450945,"CPU Wait for video finish"),(65,6442450945,"CPU Wait for GL finish"),(66,6442450945,"CPU Wait for CL finish"),(106,6442450945,"In use system memory"),(105,6442450945,"In use video memory"),(103,6442450945,"Command Queue Count"),(104,6442450945,"GPU Restart Count")),"IOReportGroupName"="Internal Statistics","IOReportChannelInfo"={"IOReportChannelUnit"=0}})
      "IOProbeScore" = 0
      "IOGVAHEVCEncodeCapabilities" = {"VTPerProfileDetails"={"1"={"VTMaxEncodeLevel"=153}},"VTQualityRating"=50,"VTRating"=350,"VTSupportedProfileArray"=(1)}
      "IOAccelDisplayPipeCapabilities" = {"DisplayPipeSupported"=Yes,"TransactionsSupported"=Yes}
      "SurfaceList" = ()
    }
    

+-o IntelAccelerator  <class IntelAccelerator, id 0x1000006f4, registered, matched, active, busy 0 (911 ms), retain 129>
    {
      "IOGeneralInterest" = "IOCommand is not serializable"
      "IODVDBundleName" = "AppleIntelKBLGraphicsVADriver"
      "IOGVAHEVCEncode" = "2"
      "InternalStatisticsAccm" = {}
      "GPURawCounterBundleName" = "AppleIntelKBLGraphicsMTLDriver"
      "MetalStatisticsName" = ("MetalStatistics","autocorr1/ComputeBasic","autocorr1/ComputeExtended","autocorr1/ComputeExtra","autocorr1/ComputeL3Cache","autocorr1/HDCAndSF","autocorr1/L3_1","autocorr1/L3_2","autocorr1/L3_3","autocorr1/MemoryReads","autocorr1/MemoryWrites","autocorr1/RasterizerAndPixelBackend","autocorr1/RenderBasic","autocorr1/RenderPipeProfile","autocorr1/Sampler","autocorr1/TDL_1","autocorr1/TDL_2")
      "IOGVAXDecode" = "2"
      "GPURawCounterPluginClassName" = "IGGPURawCounterSourceGroup"
      "InternalStatistics" = {}
      "IOGVABGRAEnc" = "Gen95"
      "AGCInfo" = {"fBusyCount"=2,"poweredOffByAGC"=No,"fSubmissionsSinceLastCheck"=14,"fLastSubmissionPID"=142}
      "MetalPluginName" = "AppleIntelKBLGraphicsMTLDriver"
      "PerformanceStatisticsAccum" = {"orphanedReusableVidMemoryHitRate"=0,"texturePageOutBytes"=0,"swapCompleteVideoWaitTime"=0,"orphanedReusableSysMemoryHitRate"=11,"dataBufferCount"=0,"oolTexturePageInBytes"=0,"textureCount"=6337,"iosurfaceTextureCreationCount"=0,"surfaceReadLockIdleWaitTime"=0,"freeDataBufferWaitTime"=0,"orphanedReusableSysMemoryBytes"=347762688,"ioSurfaceReadOutBytes"=0,"agprefTextureCreationCount"=0,"orphanedReusableSysMemoryCount"=285,"finishAll2DWaitTime"=0,"orphanedNonReusableSysMemoryCount"=0,"ioSurfacePageOutBytes"=0,"surfaceBufferReadOutBytes"=0,"surfaceTextureCreationCount"=0,"textureReadOutBytes"=0,"gartMapInBytesPerSample"=0,"swapBytesPerSample"=0,"dataBytesPerSample"=0,"surfaceCount"=20,"clientGLWaitTime"=0,"surfaceCopyInWaitTime"=0,"clientSharedAllocatedBytes"=2932736,"surfaceCopyOutWaitTime"=0,"gartMapOutBytesPerSample"=0,"ioSurfacePageInBytes"=0,"gartCacheBytes"=268435456,"volatileSurfaceCount"=0,"vramEvictionWaitTime"=0,"oolTextureCreationBytes"=0,"freeToAllocGPUAddressWaitTime"=0,"oolTextureCreationCount"=0,"orphanedReusableVidMemoryCount"=0,"orphanedReusableVidMemoryBytes"=0,"context2DCount"=1,"inUseSysMemoryBytes"=446410752,"bufferSwapCount"=0,"orphanedNonReusableVidMemoryBytes"=0,"contextGLCount"=9,"finish2DWaitTime"=0,"gartSizeBytes"=7516192768,"recoveryCount"=0,"stdTexturePageInBytes"=0,"agprefTextureCreationBytes"=0,"gartUsedBytes"=794173440,"surfaceWriteLockIdleWaitTime"=0,"surfaceBufferPageInBytes"=0,"finishGLWaitTime"=0,"iosurfaceTextureCreationBytes"=0,"finishCLWaitTime"=0,"finishVideoWaitTime"=0,"contextCLCount"=0,"freeSurfaceBackingWaitTime"=0,"swapCompleteGLWaitTime"=0,"orphanedNonReusableSysMemoryBytes"=0,"inUseVidMemoryBytes"=0,"agpTextureCreationBytes"=0,"stdTextureCreationBytes"=0,"agpTextureCreationCount"=0,"stdTextureCreationCount"=0,"surfaceSetShapeIdleWaitTime"=0,"gartFreeBytes"=6722019328,"textureVolunteerUnloadBytes"=0,"bufferFlipCount"=0,"hardwareSubmitWaitTime"=0,"swapComplete2DWaitTime"=0,"surfaceBufferPageOutBytes"=0,"surfaceBufferTextureCreationCount"=0,"hardwareWaitTime"=0,"contextVideoCount"=0,"freeSurfaceSwapBufferWaitTime"=0,"orphanedNonReusableVidMemoryCount"=0}
      "PerformanceStatistics" = {"orphanedReusableVidMemoryHitRate"=0,"texturePageOutBytes"=0,"swapCompleteVideoWaitTime"=0,"orphanedReusableSysMemoryHitRate"=11,"dataBufferCount"=0,"oolTexturePageInBytes"=27062272,"textureCount"=6337,"Device Utilization % at cur p-state"=11,"iosurfaceTextureCreationCount"=74,"freeDataBufferWaitTime"=0,"surfaceReadLockIdleWaitTime"=0,"orphanedReusableSysMemoryBytes"=347762688,"ioSurfaceReadOutBytes"=31481856,"agprefTextureCreationCount"=0,"orphanedReusableSysMemoryCount"=285,"finishAll2DWaitTime"=0,"orphanedNonReusableSysMemoryCount"=0,"ioSurfacePageOutBytes"=0,"surfaceBufferReadOutBytes"=0,"surfaceTextureCreationCount"=0,"textureReadOutBytes"=0,"gartMapInBytesPerSample"=0,"swapBytesPerSample"=0,"dataBytesPerSample"=0,"surfaceCount"=20,"Device Unit 0 Utilization %"=10,"clientGLWaitTime"=0,"clientSharedAllocatedBytes"=2932736,"surfaceCopyInWaitTime"=0,"gartMapOutBytesPerSample"=0,"ioSurfacePageInBytes"=6578176,"surfaceCopyOutWaitTime"=0,"gartCacheBytes"=268435456,"volatileSurfaceCount"=0,"vramEvictionWaitTime"=0,"oolTextureCreationBytes"=19013632,"freeToAllocGPUAddressWaitTime"=0,"oolTextureCreationCount"=89,"Device Unit 1 Utilization %"=0,"orphanedReusableVidMemoryCount"=0,"orphanedReusableVidMemoryBytes"=0,"context2DCount"=1,"inUseSysMemoryBytes"=446410752,"bufferSwapCount"=0,"orphanedNonReusableVidMemoryBytes"=0,"contextGLCount"=9,"finish2DWaitTime"=0,"Device Unit 2 Utilization %"=0,"gartSizeBytes"=7516192768,"recoveryCount"=0,"stdTexturePageInBytes"=0,"agprefTextureCreationBytes"=0,"gartUsedBytes"=794173440,"surfaceWriteLockIdleWaitTime"=0,"surfaceBufferPageInBytes"=0,"finishGLWaitTime"=0,"iosurfaceTextureCreationBytes"=28106752,"finishCLWaitTime"=0,"finishVideoWaitTime"=0,"contextCLCount"=0,"Device Unit 3 Utilization %"=0,"freeSurfaceBackingWaitTime"=0,"swapCompleteGLWaitTime"=0,"orphanedNonReusableSysMemoryBytes"=0,"inUseVidMemoryBytes"=0,"agpTextureCreationBytes"=137416704,"stdTextureCreationBytes"=0,"agpTextureCreationCount"=587,"stdTextureCreationCount"=0,"surfaceSetShapeIdleWaitTime"=0,"gartFreeBytes"=6722019328,"textureVolunteerUnloadBytes"=0,"bufferFlipCount"=0,"hardwareSubmitWaitTime"=0,"swapComplete2DWaitTime"=0,"surfaceBufferPageOutBytes"=0,"Device Utilization %"=3,"hardwareWaitTime"=23962610,"surfaceBufferTextureCreationCount"=0,"contextVideoCount"=0,"freeSurfaceSwapBufferWaitTime"=0,"orphanedNonReusableVidMemoryCount"=0}
      "IOGLBundleName" = "AppleIntelKBLGraphicsGLDriver"
      "IOPCIClassMatch" = "0x03000000&0xff000000"
      "IOSourceVersion" = "0.0.0.0.0"
      "IOPersonalityPublisher" = "com.apple.driver.AppleIntelKBLGraphics"
      "Development" = {"SignalStampRange"=1,"CsbProcModeSelect"=0,"PreemptionTimerIntervalUS"=2000,"EnableBlitLib"=1,"EUThrottle"=0,"PageTableMode"=64,"SamplerPipeThrottle"=0,"DisablePageFaultHandling"=0,"GraphicsSchedulerSelect"=0,"EnablePlaneRotation"=0,"AddressMode"=64,"DisableTailCoalescing"=0,"SemaphoreWaitPoll"=1,"SingleChannelIndex"=0,"HwCtxCacheUpdate"=1,"SurfaceSizeBasedColorBufferLLCBypass"=2,"WaitForComplete"=0,"PPGTT"=1,"AllowIOSurfaceRenderCompression"=1,"SchedSecure"=1,"AllowRenderCompression"=1,"PixelBackendThrottle"=0,"SeparateAddressSpace"=1,"TelemetryDisable"=0,"RingSizeKB"=32,"SamplerDataFetchThrottle"=0,"PreemptionDelayIntervalUS"=400,"AddressSpaceEnableMask"=3,"MetalCoalesce"=1,"ForceDisableEDRAM"=0,"PanicOnGPUHang"=0,"DisplaySurfaceInEDRAM"=2,"SchedPmNotifyEnable"=1,"MultiForceWakeSelect"=1,"NewUtilizationAlgorithmEnable"=1,"SchedThrottleCapMS"=100,"SchedulerFallbackOnFirmwareFail"=0,"EnablePreemption"=1,"EnableL3Atomics"=1,"SurfaceSizeBasedBlitDestinationBypassLLC"=2,"SignalStampsPrimary"=0}
      "IOGVAHEVCDecodeCapabilities" = {"VTPerProfileDetails"={"1"={"VTMaxDecodeLevel"=186},"2"={"VTMaxDecodeLevel"=186},"3"={"VTMaxDecodeLevel"=186}},"VTSupportedProfileArray"=(1,2,3)}
      "IOPCIPrimaryMatch" = "0x59128086 0x59168086 0x591B8086 0x591C8086 0x591E8086 0x59268086 0x59278086 0x59238086 0x87C08086 0x3E9B8086 0x3EA58086 0x3EA68086 0x3E918086 0x3E928086 0x3E988086 0x9B418086 0x9BCA8086 0x9BCC8086 0x9BC88086 0x9BC58086 0x9BC48086"
      "IOGVAH264EncodeCapabilities" = {"VTRating"=400,"VTQualityRating"=50}
      "VRAM,totalMB" = 1536
      "CFBundleIdentifier" = "com.apple.driver.AppleIntelKBLGraphics"
      "IOAccelRevision" = 2
      "IOCFPlugInTypes" = {"ACCF0000-0000-0000-0000-000a2789904e"="IOAccelerator2D.plugin"}
      "IOProviderClass" = "IOPCIDevice"
      "IOVARendererID" = 17301536
      "IOGVACodec" = "Gen95"
      "IOReportLegendPublic" = Yes
      "IOGVAHEVCDecode" = "2"
      "IOClass" = "IntelAccelerator"
      "CFBundleIdentifierKernel" = "com.apple.driver.AppleIntelKBLGraphics"
      "IOGVAScaler" = "Gen95"
      "IOReportLegend" = ({"IOReportChannels"=((4000,6442450945,"Overall GPU Busy"),(4001,6442450945,"Overall GPU total time"),(0,6442450945,"Command Buffer Bytes"),(1,6442450945,"GL Command Buffer Bytes"),(2,6442450945,"CL Command Buffer Bytes"),(3,6442450945,"2D Command Buffer Bytes"),(4,6442450945,"Video Command Buffer Bytes"),(5,6442450945,"Out-Of-Line Texture Creation Bytes"),(6,6442450945,"Shared System Memory Texture Creation Bytes"),(7,6442450945,"Private Video Memory Texture Creation Bytes"),(8,6442450945,"Private System Memory Texture Creation Bytes"),(9,6442450945,"IOSurface Backed Texture Creation Bytes"),(10,6442450945,"Out-Of-Line Texture Creation Count"),(11,6442450945,"Private Video Memory Texture Creation Count"),(12,6442450945,"Private System Memory Texture Creation Count"),(13,6442450945,"Shared System Memory Texture Creation Count"),(14,6442450945,"IOAccelSurfaceRef Texture Creation Count"),(15,6442450945,"IOAccelSurfaceBufferRef Texture Creation Count"),(16,6442450945,"IOSurface Backed Texture Creation Count"),(17,6442450945,"Texture Count"),(20,6442450945,"Bytes mapped into the GPU"),(21,6442450945,"Bytes unmapped from the GPU"),(82,6442450945,"IOAccelClientShared allocated bytes"),(22,6442450945,"Private textures bytes paged on"),(23,6442450945,"Out of line texture bytes paged on"),(24,6442450945,"IOAccelSurface bytes paged on"),(25,6442450945,"IOSurface bytes paged on"),(26,6442450945,"IOSurface bytes evicted"),(27,6442450945,"IOAccelSurface bytes evicted"),(28,6442450945,"Texture bytes evicted"),(29,6442450945,"IOSurface bytes paged off"),(30,6442450945,"IOAccelSurface bytes paged off"),(31,6442450945,"Texture bytes paged off"),(32,6442450945,"Texture bytes voluntarily paged off"),(33,6442450945,"IOAccelSurface count"),(34,6442450945,"Volatile IOAccelSurface count"),(35,6442450945,"2D context count"),(36,6442450945,"Video context count"),(37,6442450945,"OpenGL context count"),(38,6442450945,"OpenCL context count"),(39,6442450945,"Buffer swap count"),(40,6442450945,"Buffer flip count"),(41,6442450945,"Databuffer count"),(42,6442450945,"VRAM eviction wait time"),(43,6442450945,"CPU wait on GPU"),(44,6442450945,"CPU wait to submit to GPU"),(45,6442450945,"GL Client wait on GPU"),(46,6442450945,"CL Client wait on GPU"),(47,6442450945,"CPU Wait to evict resource"),(49,6442450945,"CPU wait for free databuffer"),(53,6442450945,"GL Client wait for swap complete"),(55,6442450945,"CPU Wait to copy in surface"),(56,6442450945,"CPU Wait to copy out surface"),(59,6442450945,"CPU Wait for surface idle"),(60,6442450945,"CPU Wait for resource read"),(61,6442450945,"CPU Wait for resource write"),(62,6442450945,"CPU Wait for all 2D finish"),(63,6442450945,"CPU Wait for 2D finish"),(64,6442450945,"CPU Wait for video finish"),(65,6442450945,"CPU Wait for GL finish"),(66,6442450945,"CPU Wait for CL finish"),(106,6442450945,"In use system memory"),(105,6442450945,"In use video memory"),(103,6442450945,"Command Queue Count"),(104,6442450945,"GPU Restart Count")),"IOReportGroupName"="Internal Statistics","IOReportChannelInfo"={"IOReportChannelUnit"=0}})
      "IOMatchCategory" = "IOAccelerator"
      "AccelCaps" = 15
      "IOAccelDisplayPipeCapabilities" = {"DisplayPipeSupported"=Yes,"TransactionsSupported"=Yes}
      "IOGVAHEVCEncodeCapabilities" = {"VTPerProfileDetails"={"1"={"VTMaxEncodeLevel"=156},"2"={"VTMaxEncodeLevel"=156}},"VTQualityRating"=80,"VTRating"=100,"VTSupportedProfileArray"=(1,2)}
      "IOProbeScore" = 1000
      "SurfaceList" = ({"surfaceType"="ctx","width"=64,"height"=64,"pid"=2930},{"surfaceType"="ctx","width"=64,"height"=64,"pid"=2930})
    }
    

```

@exelban

**Comment by Icarus-Xu** [comment]
yep nothing looks like "tempreture"

**Comment by exelban** [comment]
Yeap. Unfortunately, I don't see any temperature(
And do you have the temperature on Intel graphics? Because it must be unavailable too according to this output.

**Comment by Icarus-Xu** [comment]
@exelban that would be strange, I do have temp on intel graphics, and it gets updated periodically

**Comment by Icarus-Xu** [comment]
<img width="325" alt="Screen Shot 2020-11-19 at 8 15 15 PM" src="https://user-images.githubusercontent.com/11155311/99664966-f1d16580-2aa3-11eb-9d13-ab7a6901ac22.png">

**Comment by exelban** [comment]
So maybe you run this command when IG was disabled. Because recording to the output. There is no information about IG temperature.

Unfortunately, I don't find any other way to obtain GPU usage/temperature from macOS. Only nvidia/interl/amd drivers can provide this info. And if they do not provide this info, I cannot obtain it(

**Comment by Military-Doctor** [comment]
I have the same problem.

**Comment by exelban** [comment]
@Military-Doctor you can help do debug it. Just add a few prints to the GPU reader (I suppose you can run the app locally)).

**Comment by exelban** [comment]
Closed due inactivity.
